lib/runner/TestRunner.js

```diff
@@ -160,6 +168,42 @@ module.exports = class TestRunner {
         return testResults;
     }
 
+    async saveTestResults(testResults, testSuite, startTimeTestSuite) {
+        const urlString = process.env.SOURCE_API_BASE_URL ? process.env.SOURCE_API_BASE_URL : "https://source-api.bespoken.tools";
+        const urlParts = url.parse(urlString);
+        const token =  process.env.SOURCE_API_ACCESS_TOKEN || "";
+
+        const options = {
+            headers: {
+                "Content-Type": "application/json",
+                "x-access-token": token,
+            },
+            host: urlParts.hostname,
+            method: "POST",
+            path: "/v2/testResultsHistory",
+            port: urlParts.port,
+            protocol: urlParts.protocol,
+        };
+
+        const { projectId, fileName, virtualDeviceToken } = testSuite;
+        const runId = uniqid();
+
+        const body = {
+            projectId,
+            runId,
+            testResults,
+            testSuite: fileName,
+            timestamp: startTimeTestSuite,
+            token: virtualDeviceToken,
+        };
+
+        const response = await HTTP.post(options, body);
+        if (response.message.statusCode !== 200) {
+            throw new FrameworkError("Error saving test results" + " Raw Error: " + response.body.trim());
```

Comment:
Failing to save the test history should not affect the user tests, we need to fail silently hear, but we can still output the error to the bst-debug.log